src/bin/jcli_app/auto_completion.rs

```diff
@@ -12,7 +12,18 @@ pub struct AutoCompletion {
 }
 
 impl AutoCompletion {
-    pub fn exec<S: StructOpt>(self) {
-        S::clap().gen_completions("jcli", self.shell, self.output)
+    pub fn exec<S: StructOpt>(self) -> Result<(), Error> {
+        if !self.output.exists() {
+            std::fs::create_dir_all(&self.output)?;
```

Comment:
It's _very_ surprising if autocompletion makes any changes to the filesystem.
Could it just silently produce nothing if the directory does not exist?

Reply:
But I think that it's exactly the behavior that users were expecting when they were filing the bug

Reply:
ha no I think I prefer to fail if the directory does not exit. I don't want we create a directory in the user's environment. Especially they they will expect to send it in their already specific environment (for bash, in `/etc/completion.d`, zsh: `~/.zfunc`)

Reply:
yes add an error to not make it silent, but don't create the directory.

also a little update on the doc would be welcome regarding this command. It seems users are confused so would be great to make sure we have given them enough information.

Reply:
The documentation explicitly asks for directory in output parameter, that probably should be enough